base/src/main/java/com/smartdevicelink/proxy/SystemCapabilityManager.java

```diff
@@ -549,18 +550,27 @@ private void retrieveCapability(final SystemCapabilityType systemCapabilityType,
 		}
 		final GetSystemCapability request = new GetSystemCapability();
 		request.setSystemCapabilityType(systemCapabilityType);
-		request.setSubscribe((subscribe != null) ? subscribe : isSubscribedToSystemCapability(systemCapabilityType)); // If subscribe is null, then don't change the current subscription status
+		// If subscribe is null, then don't change the current subscription status
+		final boolean shouldSubscribe = ((subscribe != null) ? subscribe : isSubscribedToSystemCapability(systemCapabilityType)) && supportsSubscriptions();
```

Comment:
```suggestion
		final boolean shouldSubscribe = ((subscribe != null) ? subscribe : (isSubscribedToSystemCapability(systemCapabilityType)) && supportsSubscriptions());
```

I think the name of this is a bit misleading, because if it's `false`, it `shouldUnsubscribe()`. I almost think you need an enum with values `subscribe`, `unsubscribe`, and `keepCurrentState`. Or at least the documentation needs to be improved here.

I also need clarification on a point here. If `subscribe` is `false` and we are currently subscribed, and we support subscriptions, then `shouldSubscribe` will be `true`, right? And that's the opposite of what we want, since we want to unsubscribe?